clang/docs/RealtimeSanitizer.rst

```diff
@@ -194,12 +198,43 @@ Some issues with flags can be debugged using the ``verbosity=$NUM`` flag:
    misspelled_flag
    ...
 
-Disabling
----------
+Disabling and suppressing
+-------------------------
 
-In some circumstances, you may want to suppress error reporting in a specific scope.
+There are multiple ways to suppress error reporting when using RealtimeSanitizer.
 
-In C++, this is achieved via  ``__rtsan::ScopedDisabler``. Within the scope where the ``ScopedDisabler`` object is instantiated, all sanitizer error reports are suppressed. This suppression applies to the current scope as well as all invoked functions, including any functions called transitively.
+In general, ``ScopedDisabler`` should be preferred, as it is the most performant.
+
+.. list-table:: Suppression methods
+   :widths: 30 15 15 10 70
+   :header-rows: 1
+
+   * - Suppression method
+     - Specified at?
+     - Scope
+     - Run-time cost
+     - Description
+   * - ``ScopedDisabler``
+     - Compile-time
+     - Stack
+     - Very low
```

Comment:
I appreciate that we might want something really concise and snappy here, but if that's not a massive motivation, I think we could quantify these costs a bit, rather than just saying low, medium and high. Maybe something like "Very low: single bool check" could work?

Reply:
Yes, I'm amenable to this, but I started down this path then aborted.

My worry was providing too much low level unnecessary detail to the user, when a high level overview would suffice.

Given the info about what the actual cost is (vs "very low") how would I, the user, behave differently? Is additional (possibly complex to understand) knowledge about what is happening under the hood helpful? (honest question, I originally decided "no" but have definitely been turning this over in my head.

Reply:
Totally understood, and well phrased question - is what is happening under the hood helpful? In general it depends on the context, of course. For something like choosing between `std::map` vs `std::unordered_map` for performance reasons, it's critical to know what's happening under the hood. For choosing what json library to use in an application that isn't performance sensitive, it doesn't matter - I just care about the API being easy to use. So where does rtsan fall on that spectrum? I think probably somewhere in the middle, but I suspect leaning more towards the "I don't care" side. So maybe we just leave this as it is, and add more detail if people request it? I don't feel strongly either way on this one, just wanted to raise the question in case it rung some alarm bells.

Reply:
Yeah, I think it's good that you raised it, especially because I had the exact same back and forth. I'll leave this conversation as unresolved and maybe fmayer has an opinion.

I lean towards keeping it simple like this, but a very slight lean.

Reply:
I would also not add too many implementation details into the public documentation. It's unlikely to be useful, not unlikely to be wrong at some point in the future when the implementation changes.

Reply:
Great - agreed. I'm happy to leave as it is and for this comment to be resolved 👍